core_lib/data_transform/result_to_dict.py

```diff
@@ -100,6 +107,10 @@ def result_to_dict(return_val, properties_as_dict: bool = True, callback: Callab
     else:
         results = return_val
 
+    if isinstance(return_val, (mongomock.collection.Cursor, pymongo.cursor.Cursor)):
+        results = __pymongo_to_dict(return_val)
+        return results

```

Comment:
what is the reason for the return statement?

Reply:
It is returning the converted dict or list from result_to_dict function to where we are calling it